### PR TITLE
Enhancement: support automatically archive similar alarms

### PR DESCRIPTION
**Pull request by wzjjack**

### CODE REVIEW

File: alarm/AlarmManager2.js [hunk 1]
```diff
@@ -1220,9 +1233,6 @@ module.exports = class {
 
     let userFeedback = info.info;
```
Comment (drahnieR) [inline alarm/AlarmManager2.js]: userFeedback is a confusing name, change it to something else

File: alarm/AlarmManager2.js [hunk 2]
```diff
@@ -1602,4 +1459,167 @@ module.exports = class {
...
+              i_type = "country";
+              i_target = userFeedback.target;
+              break;
+            default:
+              break
+          }
+        }
+        break;
+    }
+    if (userFeedback && userFeedback.archiveAlarmByType) {
```
Comment (drahnieR) [inline alarm/AlarmManager2.js]: info.archiveAlarmByType & info.device are quite similar arguments but have different naming style. Make sure it's constant.

File: alarm/AlarmManager2.js (hunk 1 above)
Comment (wzjjack) [reply alarm/AlarmManager2.js]: Just keep the name same with "alarm:allow". It is more friendly when iOS developer plugin API.

File: alarm/AlarmManager2.js [hunk 3]
```diff
@@ -1602,4 +1459,167 @@ module.exports = class {
...
+      //if user archive all ALARM_DEVICE_OFFLINE
+      //only match alarm type, ignore p.device.mac,p.dest.ip, etc
+      i_type = alarm.type;
```
[the remark this answers is not in the document]
Comment (wzjjack) [reply alarm/AlarmManager2.js]: Nope. I add a new argument. User can archive all "Device Offline" alarms.

File: alarm/AlarmManager2.js [hunk 4]
```diff
@@ -248,29 +248,36 @@ module.exports = class {
...
   }
 
-  async ignoreAlarm(alarmID) {
+  async ignoreAlarm(alarmID, info) {
     log.info("Going to ignore alarm " + alarmID);
-
-    let alarm = await this.getAlarm(alarmID)
+    const userFeedback = info.info;
+    const matchAll = info.matchAll;
+    const alarm = await this.getAlarm(alarmID)
```
[the remark this answers is not in the document]
Comment (wzjjack) [reply alarm/AlarmManager2.js]: "matchAll" means it will archive some related alarms, not just one specific alarm.

**Pull request merged**